src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -116,7 +116,7 @@ private void RewriteTags(Block input, ErrorSink errorSink, int depth)
                         continue;
                     }
                 }
-                else
+                else if (!IsCommentTag((Span)child))
```

Comment:
Also suggest putting this check inside the Validate method so we have all validation in one place.